pkg/analyzer/analyzer.go

```diff
@@ -54,6 +55,14 @@ func run(pass *analysis.Pass) (interface{}, error) {
 				break
 			}
 
+			if lhs := getRootIdent(pass, assignStmt.Lhs[0]); lhs != nil {
+				if obj := pass.TypesInfo.ObjectOf(lhs); obj != nil {
+					if obj.Pos() >= body.Pos() && obj.Pos() < body.End() {
+						continue
+					}
+				}
+			}
```

Comment:
Could you add a comment explaining the reason behind this block?

Otherwise LGTM :shipit: 🚀

Reply:
Done. How's that explanation?